### PR TITLE
Handle tf.IndexedSlices types when scaling local gradients in TF.

### PR DESCRIPTION
**Pull request by romerojosh**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [x] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

This PR fixes the local gradient scaling introduced in #3719 when the gradients are sparse (e.g. of type `tf.IndexedSlices`).

### CODE REVIEW

**Comment by MrAta** [comment]
Thanks for the fix @romerojosh!
Do we also need to apply the same check in `gradient_aggregation_eager.py::_allreduce_helper` and `gradient_aggregation.py::_allreduce_grads_helper` as well?

**Comment by romerojosh** [comment]
This PR is good for review. The currently failing part of the CI is unrelated to this change and due to this issue reported by Max: https://github.com/horovod/horovod/issues/3797

File: test/parallel/test_tensorflow.py [hunk 1]
```diff
@@ -4169,7 +4169,77 @@ def test_partial_distributed_gradient_tape(self):
...
+
+        if tf.test.is_gpu_available(cuda_only=True):
+            with tf.device("/gpu:%d" % local_rank):
+                # "Transpose" input from data parallel to model parallel
+                mp_inputs = hvd.alltoall(dp_inputs)
+                mp_loss = mp_train_step(mp_inputs)
+        else:
+            # "Transpose" input from data parallel to model parallel
+            mp_inputs = hvd.alltoall(dp_inputs)
+            mp_loss = mp_train_step(mp_inputs)
```
Comment (maxhgerlach) [inline test/parallel/test_tensorflow.py:4242]: In its current form the test only asserts that the code doesn't crash, but that's better than having no test at all.

**Review by maxhgerlach — APPROVED** [review]
LGTM, thanks for the fix, @romerojosh!

I think the test case can be expanded eventually if there's further interest in developing this class of features.
